bin/handlers/pnpm-handler.ts

```diff
@@ -27,314 +20,263 @@ import {
   type ParsedOption,
 } from '../utils/text-utils.js';
 
-// regex to detect options section in help text
 const OPTIONS_SECTION_RE = /^\s*Options:/i;
+const LEVEL_MATCH_RE = /(?:levels?|options?|values?)[^:]*:\s*([^.]+)/i;
+const LINE_SPLIT_RE = /\r?\n/;
+const REPORTER_LINE_RE = /^\s*--reporter\s+\w/;
+const REPORTER_MATCH_RE = /^\s*--reporter\s+(\w+(?:-\w+)*)\s+(.+)$/;
+const SILENT_REPORTER_RE = /-s,\s*--silent,\s*--reporter\s+silent\s+(.+)/;
+const COMMA_SPACE_SPLIT_RE = /[,\s]+/;
+
+function toLines(text: string): string[] {
+  return stripAnsiEscapes(text).split(LINE_SPLIT_RE);
+}
+
+function findCommandDescColumn(lines: string[]): number {
+  let col = Number.POSITIVE_INFINITY;
+  for (const line of lines) {
+    const m = line.match(COMMAND_ROW_RE);
+    if (!m) continue;
+    const idx = line.indexOf(m[2]);
+    if (idx >= 0 && idx < col) col = idx;
+  }
+  return col;
+}
+
+function findOptionDescColumn(lines: string[], flagsOnly: boolean): number {
+  let col = Number.POSITIVE_INFINITY;
+  for (const line of lines) {
+    const m = line.match(OPTION_ROW_RE);
+    if (!m) continue;
+    if (flagsOnly && m.groups?.val) continue; // skip value-taking options in flagsOnly mode
+    const idx = line.indexOf(m.groups!.desc);
+    if (idx >= 0 && idx < col) col = idx;
+  }
+  return col;
+}
 
 function extractValidValuesFromHelp(
   helpText: string,
   optionName: string
-): string[] {
-  const lines = stripAnsiEscapes(helpText).split(/\r?\n/);
+): Array<{ value: string; desc: string }> {
+  const lines = toLines(helpText);
+
+  // edge case: reporter often appears as multiple lines
+  if (optionName === 'reporter') {
```

Comment:
we should never create edge cases that can be broken at any point. the algorithm here should be more general

imagine if somehow another option appeared like reporter, or reporter itself changed its structure. then this would break.